ddl/index.go

```diff
@@ -584,36 +597,55 @@ func (d *ddl) addTableIndex(t table.Table, indexInfo *model.IndexInfo, reorgInfo
 			err1 := kv.RunInNewTxn(d.store, true, func(txn kv.Transaction) error {
 				return errors.Trace(reorgInfo.UpdateHandle(txn, nextHandle))
 			})
-			log.Warnf("[ddl] total added index for %d rows, this task [%d,%d) add index for %d failed %v, take time %v, update handle err %v",
-				addedCount, baseHandle, nextHandle, taskAddedCount, err, sub, err1)
+			log.Warnf("[ddl] total added index for %d rows, this task [%d,%d) add index for %d failed %v, batch %d, take time %v, update handle err %v",
+				addedCount, logStartHandle, nextHandle, taskAddedCount, err, currentBatchSize, sub, err1)
 			return errors.Trace(err)
 		}
 		d.reorgCtx.setRowCountAndHandle(addedCount, nextHandle)
 		batchHandleDataHistogram.WithLabelValues(batchAddIdx).Observe(sub)
-		log.Infof("[ddl] total added index for %d rows, this task [%d,%d) added index for %d rows, take time %v",
-			addedCount, baseHandle, nextHandle, taskAddedCount, sub)
+		log.Infof("[ddl] total added index for %d rows, this task [%d,%d) added index for %d rows, batch %d, take time %v",
+			addedCount, logStartHandle, nextHandle, taskAddedCount, currentBatchSize, sub)
 
 		if isEnd {
 			return nil
 		}
-		baseHandle = nextHandle
+		baseHandle, logStartHandle = nextHandle, nextHandle
 	}
 }
 
 func getCountAndHandle(workers []*worker) (int64, int64, bool, error) {
 	taskAddedCount, nextHandle := int64(0), workers[0].taskRange.startHandle
 	var err error
 	var isEnd bool
+	lessMinWorkers := 0
+	largerDefaultWorkers := 0
 	for _, worker := range workers {
 		ret := worker.taskRet
 		if ret.err != nil {
 			err = ret.err
 			break
 		}
 		taskAddedCount += int64(ret.count)
+		if ret.count < minTaskHandledCnt {
+			lessMinWorkers++
+		} else if ret.count > defaultTaskHandleCnt {
+			largerDefaultWorkers++
+		}
 		nextHandle = ret.outOfRangeHandle
 		isEnd = ret.isAllDone
 	}
+
+	// Adjust the worker's batch size.
```

Comment:
Add more comment here to describe why we should increase the batch size. What is the problem we are trying to solve.

Reply:
Done.
PTAL @shenli